core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -190,7 +190,7 @@ final class KafkaMetadataLog private (
   override def updateHighWatermark(offsetMetadata: LogOffsetMetadata): Unit = {
     offsetMetadata.metadata.asScala match {
       case Some(segmentPosition: SegmentPosition) => log.updateHighWatermark(
-        new kafka.server.LogOffsetMetadata(
+        new org.apache.kafka.server.log.internals.LogOffsetMetadata(
```

Comment:
Why do we need to use the fully qualified name here?

Reply:
This class also uses `org.apache.kafka.raft.LogOffsetMetadata`. The argument of that method is actually of that type.

Reply:
Ok, two classes with the same name are annoying in Java. I guess we'll have to see if we can consolidate (as you said elsewhere).